### ISSUE TITLE
odata client generate invalid character on URL 

### ISSUE THREAD

**Issue by leokc-ng** [simple-odata-client/Simple.OData.Client#896]
Hi, I am having issue with the  odataclient generate additional single quote on part of my URl when I try to generate URL similar to the following example

https://testwebsite.com/api/data/v9.2/EntityDefinitions(LogicalName='account')?$select=DisplayName

I constructed the query as follow
string filterStr = $"LogicalName=\'account\'"
var oDataQuery = client.For("EntityDefinitions").Key(filterStr).Select("DisplayName"); 
var result = (await oDataQuery.FindEntryAsync());

I am getting WebRequestException where the resulting URL is 

https://testwebsite.com/api/data/v9.2/EntityDefinitions(%27LogicalName%3D%27%27aadusers%27%27%27)?$select=DisplayName

So basically I get an additional ' wrap around my key

I trace this error down to CommandFormatter ConvertValueToUriLiteral(object value, bool escapeDataString) method where  ODataUriUtils.ConvertToUriLiteral method create the additional single quote.

Please point me to the right direction if this is a bug or how I should addressed this issue.

Thanks
